### ISSUE TITLE
Add service worker static routing API information to resource timing

### ISSUE THREAD

**Issue by sisidovski** [w3c/resource-timing#389]
[Service Worker Static Routing API](https://github.com/WICG/service-worker-static-routing-api) is the API that allows developers to selectively choose whether the ServiceWorker should intercept requests, and allow them to specify when to not run ServiceWorker. In addition, it allows the them to offload simple ServiceWorker operations like cache look up. i.e. they can return resources from CacheStorage without running ServiceWorkers.

The API was proposed in [service-worker-static-routing-api](https://github.com/WICG/service-worker-static-routing-api), and it was merged into the SW spec in https://github.com/w3c/ServiceWorker/pull/1701 recently.

Service Worker provides [timing information](https://w3c.github.io/ServiceWorker/#service-worker-timing) to mark certain points in time. However, it currently does not have any fields related to the ServiceWorker Static Routing API. Developers would benefit from having fields that provide information such as matched router info or taken time to evaluate the registered SW router.

For more details, please see the static routing API explainer for the resource timing here.
https://github.com/WICG/service-worker-static-routing-api/blob/main/resource-timing-api.md

We appreciate any feedback from the community!

**Comment by nicjansma** [comment]
This was presented and discussed on the June 6th 2024 WebPerf WG call: [minutes](https://docs.google.com/document/d/1lhtCeXknwsoVzzMslJmXlYzjgZa0LE0W-ZP3UL8D7to/edit)

Summary of some points in the discussion:

* `fetchStart` would be best aligned to `responseStart` (rather than `0`) for compat issues
* Some of the questions/concerns here are similar to issues discussed in:
    * https://github.com/w3c/navigation-timing/issues/199
    * https://github.com/w3c/resource-timing/issues/363
* Can we prefix e.g. `finalRouterSource` with `workerFinalRouterSource` since the others are also prefixed?

The working group is aiming to schedule a follow-up meeting to align on all of the above.

**Comment by quasi-mod** [comment]
FYI, we’ve started the Chrome origin trial from M131 to add the proposed fields to resource timing api. 
If you are interested, please participate from the origin trial dashboard [link](https://developer.chrome.com/origintrials/#/view_trial/1689412810217357313).
We would really appreciate your insightful feedback!

**Comment by quasi-mod** [comment]
@ErikWitt Thanks for providing us with feedback for SW static routing API! Based on your interest, I thought you 
might be interested in this Origin Trial as well. Let us know what you think.

**Comment by ErikWitt** [comment]
Thanks for the heads up. I share some first feedback and a question [here](https://github.com/WICG/service-worker-static-routing-api/issues/35)

**Comment by davehunt** [comment]
Tracking Mozilla's position in https://github.com/mozilla/standards-positions/issues/1031

**Comment by yoavweiss** [comment]
@sisidovski @quasi-mod - Are there any spec changes required from Resource Timing for this? Or were all the changes done in Service Workers?

**Comment by sisidovski** [comment]
@quasi-mod Can you check the current status? AFAIK these are all spec changes, and we don't need to change Resource Timing spec, correct?

- https://github.com/w3c/ServiceWorker/pull/1769
- https://github.com/whatwg/fetch/pull/1832

**Comment by yoavweiss** [comment]
> * Add new fields and getter steps for static routing resource timing ServiceWorker#1769

I see the "service worker timing info" defined here, but I don't see where those timestamps are exposed..

**Comment by yoavweiss** [comment]
Yes, but I don't see an actual IDL that defines an API that exposes this to web developers. (the one [implemented here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/timing/performance_resource_timing.idl;l=45?q=worker%20f:resource%20f:idl$&ss=chromium%2Fchromium%2Fsrc))
